### PR TITLE
shorten e2e_node test, fix pod ready test issue

### LINKED ISSUE #25703: Node E2E: Make conformance test run faster
Slow conformance test has caused issues like #25639 

@liangchenye We should make runtime/image conformance test run faster without breaking the test coverage, possibly by:
* Shorten consistently check interval #25695.
* Remove redundant test cases such as https://github.com/kubernetes/kubernetes/pull/24191#discussion_r62968819.
* ...

/cc @pwittrock

### PR DESCRIPTION
**Pull request by liangchenye**
Merge test cases as @Random-Liu suggested.
fix #25703

Signed-off-by: liang chenye liangchenye@huawei.com

### CODE REVIEW

**Comment by Random-Liu** [comment]
Also fix #25812

File: test/e2e_node/runtime_conformance_test.go [hunk 1]
```diff
@@ -85,10 +85,9 @@ var _ = Describe("Container runtime Conformance Test", func() {
 						},
 					},
 				}
-				testCount := int32(3)
 				testStatuses := []testStatus{
-					{"terminate-cmd-rpa", api.RestartPolicyAlways, api.PodRunning, ContainerStateWaiting | ContainerStateRunning | ContainerStateTerminated, ">", testCount, false},
-					{"terminate-cmd-rpof", api.RestartPolicyOnFailure, api.PodSucceeded, ContainerStateTerminated, "==", testCount, false},
+					{"terminate-cmd-rpa", api.RestartPolicyAlways, api.PodRunning, ContainerStateWaiting | ContainerStateRunning | ContainerStateTerminated, "==", 2, true},
```
Comment (Random-Liu) [inline test/e2e_node/runtime_conformance_test.go]: Why do we still need `ContainerStateWaiting | ContainerStateRunning | ContainerStateTerminated`?
The container must be running, right?

File: test/e2e_node/runtime_conformance_test.go [hunk 2]
```diff
@@ -85,10 +85,9 @@ var _ = Describe("Container runtime Conformance Test", func() {
...
+					{"terminate-cmd-rpof", api.RestartPolicyOnFailure, api.PodSucceeded, ContainerStateTerminated, "==", 1, false},
 					{"terminate-cmd-rpn", api.RestartPolicyNever, api.PodSucceeded, ContainerStateTerminated, "==", 0, false},
```
Comment (Random-Liu) [inline test/e2e_node/runtime_conformance_test.go]: I believe the pod will be `api.PodFailed`.

**Comment by liangchenye** [comment]
@Random-Liu I tried several times and still see the 'ready' issue.

[commit 0d0d6f8 is my final commit, but it fails in the first pod test]
this pod restarts twice and then keeps running, but the 'ready' status is always 'false'.

[commit 9bb07bd is an experimental commit]
I only keep one pod which  keeps running, the 'ready' status could be 'true' (as expected).

It seems: a pod could never be ready if it restarts several times.
I did not find a document talking about this, is this a bug of k8s?

**Comment by Random-Liu** [comment]
@liangchenye Yeah, it's wired. I was also investigating that yesterday when reviewing the code.
I'll look into it.

File: test/e2e_node/runtime_conformance_test.go [hunk 3]
```diff
@@ -98,7 +95,19 @@ var _ = Describe("Container runtime Conformance Test", func() {
 					defer os.Remove(tmpFile.Name())
 
 					// It fails in the first three runs and succeeds after that.
-					tmpCmd := fmt.Sprintf("echo 'hello' >> /restart-count/%s ; test $(wc -l /restart-count/%s| awk {'print $1'}) -ge %d", path.Base(tmpFile.Name()), path.Base(tmpFile.Name()), testCount+1)
+					cmdScripts := `
+f=%s
+count=$(echo 'hello' >> $f ; wc -l $f | awk {'print $1'})
```
Comment (Random-Liu) [inline test/e2e_node/runtime_conformance_test.go]: change to /restart-count/$f
That's the reason.

**Comment by Random-Liu** [comment]
Close this one. The new one is at https://github.com/kubernetes/kubernetes/pull/26856

**Pull request closed without merging**